lib/puma/cluster.rb

```diff
@@ -35,34 +35,19 @@ def stop_workers
       @workers.each { |x| x.term }
 
       begin
-        if RUBY_VERSION < '2.6'
-          @workers.each do |w|
+        pids = @workers.map(&:pid)
+        loop do
+          pids.reject! do |w_pid|
             begin
-              Process.waitpid(w.pid)
-            rescue Errno::ECHILD
-              # child is already terminated
-            end
-          end
-        else
-          # below code is for a bug in Ruby 2.6+, above waitpid call hangs
-          t_st = Process.clock_gettime(Process::CLOCK_MONOTONIC)
-          pids = @workers.map(&:pid)
-          loop do
-            pids.reject! do |w_pid|
-              begin
-                if Process.waitpid(w_pid, Process::WNOHANG)
-                  log "    worker status: #{$?}"
-                  true
-                end
-              rescue Errno::ECHILD
-                true # child is already terminated
+              if Process.wait(w_pid, Process::WNOHANG)
+                true
               end
+            rescue Errno::ECHILD
+              true # child is already terminated
             end
-            break if pids.empty?
-            sleep 0.5
           end
-          t_end = Process.clock_gettime(Process::CLOCK_MONOTONIC)
```

Comment:
not a fan of logging? 😆

Reply:
I don't know.  I think I added that recently.  I can log/debug lines if you'd like...